lib/api/api.js

```diff
@@ -146,6 +146,16 @@ const api = {
 
         const requestContexts = prepareRequestContexts(apiMethod, request,
             sourceBucket, sourceObject, sourceVersionId);
+
+        if (apiMethod === 'completeMultipartUpload' || apiMethod === 'multipartDelete') {
+            // Request account quotas explicitly for MPU requests, to consider parts cleanup
+            // NOTE: we need quota for these, but it will be evaluated at the end of the API,
+            // once the parts have actually been deleted (not via standardMetadataValidateBucketAndObj)
+            requestContexts.forEach(context => {
+                context._needQuota = true; // eslint-disable-line no-param-reassign
+            });
+        }
```

Comment:
should this not be added inside `prepareRequestContexts`, since it has all the different cases for each api method?